core/src/main/java/org/javarosa/core/model/SelectChoice.java

```diff
@@ -121,4 +121,46 @@ public String getTextID() {
     public void setTextID(String textID) {
         this.textID = textID;
     }
+
+    @Override
+    public int hashCode() {
+        int result;
+        result = textID != null ? textID.hashCode() : 0;
+        result = 31 * result + value.hashCode();
+        result = 31 * result + index;
```

Comment:
Yea, was honestly just copying the structure of sample implementations. Think you're right about xor though, can add that.